csrc/ops/composite.cpp

```diff
@@ -546,48 +560,39 @@ SdpfaBwdResult sdpfa_bwd(
     TensorView* philox_seed,
     TensorView* philox_offset,
     Val* scale) {
-  NVF_CHECK(
-      query->dtype() == key->dtype() && query->dtype() == value->dtype(),
-      "Expected query, key, and value to have the same dtype but got: ",
-      query->dtype(),
-      " ",
-      key->dtype(),
-      " ,and ",
-      value->dtype());
-
+  checkAllEqual(
```

Comment:
This and the checkAllEqual at line 576 will harden the checks. Currently, the code checks only Q, K and V.